packages/paypal-commerce-integration/src/paypal-commerce-fastlane/paypal-commerce-fastlane-payment-strategy.ts

```diff
@@ -133,22 +133,33 @@ export default class PaypalCommerceFastlanePaymentStrategy implements PaymentStr
         const state = this.paymentIntegrationService.getState();
         const cartId = state.getCartOrThrow().id;
 
-        const { orderId } = await this.paypalCommerceRequestSender.createOrder(methodId, {
-            cartId,
-        });
+        const isVaultedFlow = paymentData && isVaultedInstrument(paymentData);
+
+        try {
+            const { id = '' } = isVaultedFlow ? {} : await this.getPayPalFastlanePaymentToken();
+
+            const { orderId } = await this.paypalCommerceRequestSender.createOrder(methodId, {
+                cartId,
+            });
 
-        const paymentPayload =
-            paymentData && isVaultedInstrument(paymentData)
+            const paymentPayload = isVaultedFlow
                 ? this.prepareVaultedInstrumentPaymentPayload(methodId, orderId, paymentData)
-                : await this.preparePaymentPayload(methodId, orderId, paymentData);
+                : this.preparePaymentPayload(methodId, orderId, paymentData, id);
 
-        await this.paymentIntegrationService.submitOrder(order, options);
-        await this.paymentIntegrationService.submitPayment<PayPalFastlanePaymentFormattedPayload>(
-            paymentPayload,
-        );
+            await this.paymentIntegrationService.submitOrder(order, options);
+            await this.paymentIntegrationService.submitPayment<PayPalFastlanePaymentFormattedPayload>(
+                paymentPayload,
+            );
 
-        // TODO: we should probably update this method with removeStorageSessionId for better reading experience
-        this.paypalCommerceFastlaneUtils.updateStorageSessionId(true);
+            // TODO: we should probably update this method with removeStorageSessionId for better reading experience
+            this.paypalCommerceFastlaneUtils.updateStorageSessionId(true);
+        } catch (error) {
+            if (error instanceof Error && error.name !== 'FastlaneError') {
```

Comment:
Maybe move this string to some constant in case we'll need it somewhere else. (optinal)

Reply:
I think it is not an appropriate case for this action, would be better to leave it as it is